[html] Specify unique names for tests

jugglinmike opened the pull request:
Duplicated test names make interpreting results difficult for humans and machines. In order to avoid this confusion, [an upcoming extension to the test harness](https://github.com/w3c/testharness.js/pull/240) will cause any test file with duplicate test names to be interpreted as an error.

jugglinmike committed:
[html] Specify unique names for tests

w3c-bots commented:
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/5049)

w3c-bots commented:
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/208254819)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 8c2b5bc4ac61fd0813f367e3cddef877363bae7f
Using browser at version BuildID 20170306110339; SourceStamp 966464a68a2cb3ca1125808e34abb5c1d34e3797
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5049/html/browsers/history/the-location-interface/location-protocol-setter-with-colon.sub.html">/html/browsers/history/the-location-interface/location-protocol-setter-with-colon.sub.html</a></summary>


|                   Subtest                   | Results | Messages |
|---------------------------------------------|---------|----------|
|                                             | OK      |          |
| `General setup`                             | PASS    |          |
| `Set location.protocol = location.protocol` | PASS    |          |
| `Set location.protocol to http:gunk`        | PASS    |          |
| `Message listener: existing`                | PASS    |          |
| `Message listener: http-and-gunk`           | PASS    |          |

</details>
</details>

w3c-bots commented:
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/208254823)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 8c2b5bc4ac61fd0813f367e3cddef877363bae7f
Using browser at version 58.0.3026.3 dev
Starting 10 test iterations
## Unstable results ##

jugglinmike committed:
fixup! [html] Specify unique names for tests

jugglinmike committed:
[html] Completely disable test

The test documents the case using the HTTPS protocol as unimplemented.
Despite this, it previously included an iframe whose `src` specified an
HTTPS URL, and this triggered the creation of a test.

Disable the iframe with an HTML comment to avoid this behavior while
maintaining consistency in the treatment of "to be implemented" test
logic.

w3c-bots commented:
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/5049)
